autometa/common/external/download_dataset.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+COPYRIGHT
+Copyright 2021 Ian J. Miller, Evan R. Rees, Kyle Wolf, Siddharth Uppal,
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+pulling data from google drive folder with simulated or synthetic communities
+"""
+
+
+import gdown
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def download_dataset(dataset, out_dirpath):
+    # provide list of database options as a dictionary with file_ids from google
+    simulated = {
+        "test": "1fy3M7RnS_HGSQVKidCy-rAwXuxldyOOv",
+        "78": "15CB8rmQaHTGy7gWtZedfBJkrwr51bb2y",
+        "156": "13bkwFBIUhdWVWlAmVCimDODWF-7tRxgI",
+        "312": "1qyAu-m6NCNuVlDFFC10waOD28j15yfV-",
+        "625": "1FgMXSD50ggu0UJbZd1PM_AvLt-E7gJix",
+        "1250": "1KoxwxBAYcz8Xz9H2v17N9CHOZ-WXWS5m",
+        "2500": "1wKZytjC4zjTuhHdNUyAT6wVbuDDIwk2m",
+        "5000": "1IX6vLfBptPxhL44dLa6jePs-GRw2XJ3S",
+        "10000": "1ON2vxEWC5FHyyPqlfZ0znMgnQ1fTirqG",
+    }
+
+    # construct file id into a url to put into gdown
+    file_id = simulated[dataset]
+    url = f"https://drive.google.com/uc?id={file_id}"
+    filename = f"{dataset}_metagenome.fna.gz"
+    out_fpath = os.path.join(out_dirpath, filename)
```

Comment:
Currently this only allows us to download each dataset's metagenome fasta file. Can we add functionality here to download any of the available files for a particular dataset? In terms of benchmarking, this functionality would be really helpful/useful. For example, if we are testing a number of different parameter sets for binning, it would be convenient to benchmark these against the existing binning file from Autometa v1.0 (available on google drive) as well as retrieve the ground-truth assignments (also available on google drive) to determine benchmarking metrics.

I've briefly described some example usage in the benchmarking pull request (https://github.com/KwanLab/Autometa/pull/141#issuecomment-847348201) detailing the usage within a nextflow process. Can we implement this functionality?

### Suggested change

Change the arguments to the function definition

```python
# From
def download(dataset: str, out_dirpath: str):
    ...
# To
def download(dataset: str, file: str, out: str) -> str:
    # 1. lookup dataset
    # 2. lookup file in dataset
    # 3. Write file to out
    ...
    return out
```

Reply:
I've just added new ability to download entire folders, specific files, or all files with the same name across folders. I haven't figured out Nextflow enough to say whether or not this will play nice there. I'm trying to just do one thing at a time! Does it look a bit more useful now?